src/libproc_macro/bridge/client.rs

```diff
@@ -340,6 +340,7 @@ pub struct Client<F> {
 
 // FIXME(#53451) public to work around `Cannot create local mono-item` ICE,
 // affecting not only the function itself, but also the `BridgeState` `thread_local!`.
+#[allow(improper_ctypes)]
```

Comment:
Does the lint only complain about this because of the Rust-ABI function pointer `f`? I don't know libproc_macro at all so idk whether this is a false positive or not, cc @eddyb

Reply:
It is a false positive, that function pointer is to the `#[proc_macro]` functions the user writes in a proc macro crate, they can't be other ABIs.
See the comment on `Client` itself (I think).

Reply:
Unless anyone disagrees, I think the correct course of action is to leave this function as is with the `#[allow(improper_ctypes)]` attribute.

Reply:
Any chance the lint could be more fine-grainedly applied to the offending argument?
Oh and technically this could take a raw pointer, or `usize`, since just passing that value around is what is.... oooh wait, I have an idea.

The functions could.be defined to take `impl Fn(A) -> R + Copy + 'static` (and other bounds necessary, if any) instead of `fn(A) -> R` - since all they need is for *some* callable to be passed (there is no `unsafe` code involved), it just happens to be a function pointer from the `Client` in practice.

Reply:
I think we could leave this as a follow-up.